config/config.go

```diff
@@ -244,12 +245,19 @@ func loadExtensions(v *viper.Viper, factories map[string]extension.Factory) (con
 		extensionCfg.SetType(typeStr)
 		extensionCfg.SetName(fullName)
 
-		// Now that the default config struct is created we can Unmarshal into it
-		// and it will apply user-defined config on top of the default.
-		if err := subViper.UnmarshalKey(key, extensionCfg); err != nil {
-			return nil, &configError{
-				code: errUnmarshalError,
-				msg:  fmt.Sprintf("error reading settings for extension type %q: %v", typeStr, err),
+		// Unmarsh only the subconfig for this extension.
+		sv := subViper.Sub(key)
+		if sv != nil {
```

Comment:
What if `sv==nil`? I think there is a need for error handling here. I would expect us to already have a test case that covers this, if we don't please consider adding.

Reply:
It is not an error. `Sub` returns nil if the config is empty:

```
exporters
  foo:
```

The `Sub` for `foo` will be nil. Before we used `UnmarshalKey` on the `exporters` sub-config which took care of this.

Reply:
Sounds good.